features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/ActiveCallManager.kt

```diff
@@ -186,28 +188,35 @@ class DefaultActiveCallManager @Inject constructor(
 
     @OptIn(ExperimentalCoroutinesApi::class)
     private fun observeRingingCall() {
-        // This will observe ringing calls and ensure they're terminated if the room call is cancelled
+        // This will observe ringing calls and ensure they're terminated if the room call is cancelled or if the user
+        // has joined the call from another session.
         activeCall
             .filterNotNull()
             .filter { it.callState is CallState.Ringing && it.callType is CallType.RoomCall }
             .flatMapLatest { activeCall ->
                 val callType = activeCall.callType as CallType.RoomCall
-                // Get a flow of updated `hasRoomCall` values for the room
+                // Get a flow of updated `hasRoomCall` and `activeRoomCallParticipants` values for the room
                 matrixClientProvider.getOrRestore(callType.sessionId).getOrNull()
                     ?.getRoom(callType.roomId)
                     ?.roomInfoFlow
-                    ?.map { it.hasRoomCall }
+                    ?.map {
+                        it.hasRoomCall to (callType.sessionId in it.activeRoomCallParticipants)
+                    }
                     ?: flowOf()
             }
             // We only want to check if the room active call status changes
             .distinctUntilChanged()
             // Skip the first one, we're not interested in it (if the check below passes, it had to be active anyway)
             .drop(1)
-            .onEach { roomHasActiveCall ->
+            .onEach { (roomHasActiveCall, userIsInTheCall) ->
                 if (!roomHasActiveCall) {
                     // The call was cancelled
                     timedOutCallJob?.cancel()
-                    incomingCallTimedOut()
+                    incomingCallTimedOut(displayMissedCallNotification = true)
+                } else if (userIsInTheCall) {
+                    // The user joined the call from another session
+                    timedOutCallJob?.cancel()
+                    incomingCallTimedOut(displayMissedCallNotification = false)
```

Comment:
Note that if the call is not ringing anymore we will run this onEach block because we are still observing the roomInfoFlow, but there is some filtering on `incomingCallTimedOut` that saves us.